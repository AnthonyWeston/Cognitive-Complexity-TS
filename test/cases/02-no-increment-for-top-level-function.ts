function f() {
    while (true);
}

const g = () => {
    while (true);
};

namespace N {
    function f() {
        while (true);
    }
}

class C {
    // depth 0
    f = () => {
        // depth 0
        while (true); // +1
    }

    g() {
        // depth 0
        while (true); // +1
    }
}