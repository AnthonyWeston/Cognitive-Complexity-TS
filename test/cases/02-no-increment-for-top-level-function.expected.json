{
    "02-no-increment-for-top-level-function.ts": {
        "score": 5,
        "inner": [
            {
                "name": "f",
                "score": 1,
                "line": 1,
                "column": 1
            },
            {
                "name": "g",
                "score": 1,
                "line": 5,
                "column": 1
            },
            {
                "name": "N",
                "score": 1,
                "line": 9,
                "column": 1,
                "inner": [
                    {
                        "name": "f",
                        "score": 1,
                        "line": 10,
                        "column": 5
                    }
                ]
            },
            {
                "name": "C",
                "score": 2,
                "line": 15,
                "column": 1,
                "inner": [
                    {
                        "name": "f",
                        "score": 1,
                        "line": 17,
                        "column": 5
                    },
                    {
                        "name": "g",
                        "score": 1,
                        "line": 22,
                        "column": 5
                    }
                ]
            }
        ]
    }
}